test/common/unit/AxelarAdapter.t.sol

```diff
@@ -169,12 +172,31 @@ contract AxelarAdapterTest is Test {
         assertEq(call[0], 0.1 ether);
         assertEq(axelarGasService.values_address("sender"), address(adapter));
         assertEq(axelarGasService.values_string("destinationChain"), AXELAR_CHAIN_ID);
-        assertEq(axelarGasService.values_string("destinationAddress"), makeAddr("DestinationAdapter").toString());
+        assertEq(axelarGasService.values_string("destinationAddress"), makeAddr("DestinationAdapter").toAxelarString());
         assertEq(axelarGasService.values_bytes("payload"), payload);
         assertEq(axelarGasService.values_address("refundAddress"), refund);
 
         assertEq(axelarGateway.values_string("destinationChain"), AXELAR_CHAIN_ID);
-        assertEq(axelarGateway.values_string("contractAddress"), makeAddr("DestinationAdapter").toString());
+        assertEq(axelarGateway.values_string("contractAddress"), makeAddr("DestinationAdapter").toAxelarString());
         assertEq(axelarGateway.values_bytes("payload"), payload);
     }
 }
+
+// From https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/contracts/libs/AddressString.sol#L30C26-L45C6
+library AxelarAddressToString {
+    function toAxelarString(address address_) internal pure returns (string memory) {
+        bytes memory addressBytes = abi.encodePacked(address_);
+        bytes memory characters = "0123456789abcdef";
+        bytes memory stringBytes = new bytes(42);
+
+        stringBytes[0] = "0";
+        stringBytes[1] = "x";
+
+        for (uint256 i; i < 20; ++i) {
+            stringBytes[2 + i * 2] = characters[uint8(addressBytes[i] >> 4)];
+            stringBytes[3 + i * 2] = characters[uint8(addressBytes[i] & 0x0f)];
+        }
+
+        return string(stringBytes);
+    }
+}
```

Comment:
Q: If this is not used in production, do we really need this for testing? 

I think we can use `vm.toString(address)`

Reply:
It is what Axelar uses in production. Just in our code, we were doing `string(abi.encodePacked(addr))` and that wasn't equivalent to Axelar's usage of toString. See Cantina Discord thread.